src/components/categories/View.vue

```diff
@@ -0,0 +1,211 @@
+<script setup lang="ts">
+import { useAsyncState } from '@vueuse/core'
+import { computed, ref } from 'vue'
+import { useApi } from '../../composables/useApi'
+import { useAuth } from '../../composables/useAuth'
+import { useSessionToken } from '../../composables/useSessionToken'
+import { Categories } from '../../lib/api/Categories'
+import { ProblemDetails } from '../../lib/api/data-contracts'
+import { HttpResponse } from '../../lib/api/http-client'
+
+const props = defineProps<{
+	id: number
+}>()
+
+const updateError = ref('')
+
+const token = useSessionToken()
+
+const frontendUrl = import.meta.env.VITE_FRONTEND_URL
+
+const categories = new Categories({
+	baseUrl: import.meta.env.VITE_BACKEND_URL
+})
+
+const {
+	state: category,
+	error,
+	isLoading,
+	execute
+} = useAsyncState(async () => {
+	const resp = await categories.getCategory(props.id)
+	return resp.data
+}, null)
+
+const errorResponse = computed(
+	() => (error.value as HttpResponse<unknown, ProblemDetails>).error
+)
+
+async function revealDelete() {
+	if (
+		confirm('Really delete this category? (This action can be reversed)')
+	) {
+		useApi(
+			() => categories.deleteCategory(props.id, useAuth(token.value)),
+			() => execute(),
+			(error) => {
+				updateError.value = 'Failed to delete: ' + error.status
+			}
+		)
+	}
+}
+
+async function revealRestore() {
+	if (
+		confirm('Really restore this category? (This action can be reversed)')
+	) {
+		useApi(
+			() => categories.restoreCategory(props.id, useAuth(token.value)),
+			() => execute(),
+			(error) => {
+				updateError.value = 'Failed to restore: ' + error.status
+			}
+		)
+	}
+}
+</script>
+
+<template>
+	<div class="container">
+		<div v-if="isLoading">Loading...</div>
+		<div v-else-if="error" class="error-container">
+			<p class="errorText">
+				Failed to fetch category: {{ errorResponse.status }}
+				{{ errorResponse.title }}
+			</p>
+			<button @click="execute()" class="button">Retry</button>
+		</div>
+
+		<div v-else class="main-content">
+			<h1 class="title">Details for {{ category?.name }}</h1>
```

Comment:
The LB View page notably lacks this heading. We should synchronize them for consistency. The category name is listed in the table so you could make this text just say "Category Details" or something generic like that. If you are in favor of adding a similar heading to the LB page, go ahead and resolve this comment and make an issue for it.

Reply:
I have added a similar title in the LB View Page already. And I was originally going to do the generic "Category Details" title, but decided that printing the category name flows better.

Reply:
In a perfect world I think there would be some sort of indication that could tell you at a glance this is the category details page. Currently, the URL is the only indication of that. I don't mind leaving it as it is now but it's something we'll keep in mind going forward maybe.